content/courses/program-optimization/rust-macros.md

```diff
@@ -587,18 +587,37 @@ to be used as a derive macro that can be applied to a struct. The line
 that this is a derive macro that processes `account` and `instruction` helper
 attributes.
 
+The INIT_SPACE is used to calculate the initial size of an account. It is
+implemented by derive macro on `MyData` automatically implementing the
+[anchor_lang::Space](https://docs.rs/anchor-lang/latest/anchor_lang/trait.Space.html#associatedconstant.INIT_SPACE).
+
+```rust
+#[account]
+#[derive(InitSpace)]
+pub struct NewAccount {
+ data: u64,
+}
+```
+
+The `#[account]` macro also automatically derives the _DISCRIMINANT_ of an
+anchor account which implements the
+[anchor_lang::Discriminator](https://docs.rs/anchor-lang/latest/anchor_lang/trait.Discriminator.html)
+trait. This trait exposes an array of 8 bytes containing the discriminator,
+which can be exposed using `NewAccount::DISCRIMINATOR`. Calling the `.len()` on
+this array of 8 bytes gives us the length of the discriminator;
+
 ```rust
 #[proc_macro_derive(Accounts, attributes(account, instruction))]
 pub fn derive_anchor_deserialize(item: TokenStream) -> TokenStream {
-    parse_macro_input!(item as anchor_syn::AccountsStruct)
+	parse_macro_input!(item as anchor_syn::AccountsStruct)
```

Comment:
Old indent was better. 4 spaces per rustfmt then 4 more for the method chaining.